src/app/shared/components/template/components/task-progress-bar/task-progress-bar.component.ts

```diff
@@ -1,81 +1,175 @@
+    private campaignService: CampaignService,
+    private cdr: ChangeDetectorRef,
+    private dynamicDataService: DynamicDataService
   ) {
     super();
   }
 
-  ngOnInit() {
+  async ngOnInit() {
     this.getParams();
-    this.evaluateTaskGroupData();
+    if (this.params.useDynamicData) {
+      this.subscribeToData();
+    } else {
+      const taskGroupDataRows = await this.getTaskGroupDataRows();
+      this.evaluateTaskGroupData(taskGroupDataRows);
+    }
   }
 
   getParams() {
+    // If component is being explicitly instantiated from a template, get the params from the template row
     if (this._row) {
-      this.taskGroupDataList = getStringParamFromTemplateRow(this._row, "task_group_data", null);
-      this.taskGroupCompletedField = getStringParamFromTemplateRow(
+      this.params.taskGroupDataList = getStringParamFromTemplateRow(
+        this._row,
+        "task_group_data",
+        null
+      );
+      this.params.taskGroupCompletedField = getStringParamFromTemplateRow(
         this._row,
         "completed_field",
         null
       );
-      this.progressUnitsName = getStringParamFromTemplateRow(
+      this.params.progressUnitsName = getStringParamFromTemplateRow(
         this._row,
         "progress_units_name",
         "sections"
       );
+      this.params.showText = getBooleanParamFromTemplateRow(this._row, "show_text", false);
+      this.params.useDynamicData = getBooleanParamFromTemplateRow(this._row, "dynamic", false);
+      this.params.variant = getStringParamFromTemplateRow(this._row, "variant", "")
+        .split(",")
+        .join(" ") as ITaskProgressBarParams["variant"];
+    }
+    // If component is being instantiated by a parent component (e.g. task-card), use Input() values for params.
+    else {
+      this.params.taskGroupDataList = this.taskGroupDataList;
+      this.params.taskGroupCompletedField = this.taskGroupCompletedField;
+      this.params.progressUnitsName = this.progressUnitsName;
+      this.params.showText = this.showText;
     }
   }
 
   get progressPercentage() {
     return (this.subtasksCompleted / this.subtasksTotal) * 100;
   }
 
-  async evaluateTaskGroupData() {
-    const dataList = await this.appDataService.getSheet("data_list", this.taskGroupDataList);
-    const subtasks = dataList?.rows || [];
+  async getTaskGroupDataRows() {
+    const dataList = await this.appDataService.getSheet("data_list", this.params.taskGroupDataList);
+    return dataList?.rows;
+  }
+
+  /**
+   * Fetch the completion status for each subtask of the task group
+   * and calculate the task group's completion percentage to display.
+   * Task group data lists can store their completion status in one of two ways:
+   * 1. A "completed_field" column, referencing a field that stores the completion status
+   * 2. A "completed" column, with values updated dynamically
+   * @param dataRows The data rows of the task group
+   */
+  async evaluateTaskGroupData(dataRows: any[]) {
```

Comment:
nit(blocking) - this method in particular would probably be better if possible to split heavy lifting to task service with tests added